### PR TITLE
Add a FrontBus to which low-bandwidth bus masters attach

### PR DESCRIPTION
**Pull request by hcook**
"Front" means in front of the system port.

Also added a bunch of `suggestNames` for modules generated as attachment points to the SystemBus. We should probably revisit how this is handled, and do it more programmatically.

### CODE REVIEW

File: src/main/scala/coreplex/SystemBus.scala [hunk 1]
```diff
@@ -20,61 +20,72 @@ case object SystemBusParams extends Field[SystemBusParams]
...
-  pbusBuffer0.node :*= pbus_fixer.node
-  pbusBuffer1.node :*= pbusBuffer0.node
-  pbusBuffer2.node :*= pbusBuffer1.node
-  pbusBuffer3.node :*= pbusBuffer2.node
-  val toPeripheryBus: TLOutwardNode = pbusBuffer3.node
+  def toPeripheryBus(nBuffers: Int): TLOutwardNode = {
+    val (in, out) = bufferChain(nBuffers, name = Some("PeripheryBus"))
+    in := pbus_fixer.node
+    out
+  }
```
Comment (terpstra) [inline src/main/scala/coreplex/SystemBus.scala]: This should be nBuffers-1. There is already a buffer before the pbus_fixer.

**Comment by terpstra** [comment]
I'm still pretty skeptical about these multiple buffer chains. Adding one buffer strategically here or there helps. I have yet to see adding more than one buffer improve timing in our designs...

**Comment by mwachs5** [comment]
I agree, but still think TLBufferChain is a useful primitive to have

**Comment by henrystyles** [comment]
Putting multiple 'AXI register slice' IPs back to back is pretty common for FPGA designs that I've seen.

**Pull request merged**